### PR TITLE
Add Promise.try() support for Firefox 132

### PR DESCRIPTION
**Pull request by chrisdavidmills**
#### Summary

[`Promise.try()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/try) is supported in Firefox as of version 132. See https://bugzilla.mozilla.org/show_bug.cgi?id=1905364 for implementation details.

#### Related issues

Main project issue: https://github.com/mdn/content/issues/36118

### CODE REVIEW

**Comment by Josh-Cena** [comment]
I'm not sure if this is the right way to record it, considering it's currently in nightly and behind a flag, based on https://phabricator.services.mozilla.com/D218494. The flag is `enable-promise-try`. I think the rule of thumb is that changing a Firefox JS feature going from "none" to "stable unflagged" is almost always wrong.

**Review by caugner — APPROVED** [review]

**Review by caugner — APPROVED** [review]
Phew, good catch, @Josh-Cena.

(There was no obvious indication on the bug itself that this was implemented behind a pref, other than the diff.)

**Comment by chrisdavidmills** [comment]
> I'm not sure if this is the right way to record it, considering it's currently in nightly and behind a flag, based on https://phabricator.services.mozilla.com/D218494. The flag is `enable-promise-try`, or `javascript.options.experimental.promise_try` in `about:config`, which is off by default. I think the rule of thumb is that changing a Firefox JS feature from "none" to "stable unflagged" is almost always wrong.

Ah, thanks @Josh-Cena. I didn't realize this. I've added the pref information.

I'm still not 100% sure this data is correct yet. From testing, it looks like the pref is available to enable and test the feature in Nightly (133), but not Beta (132). Do I update the version to 133, or is there some way of saying "nightly, whatever version that is"?

**Comment by Josh-Cena** [comment]
My experience (with JS features, at least) is that "Implement XXX" bugs are always nightly and/or flagged, and "Ship XXX" bugs are when we really add stable browser versions. I don't know about BCD conventions to record this expr status though.

File: javascript/builtins/Promise.json [hunk 1]
```diff
@@ -605,7 +612,7 @@
               "webview_ios": "mirror"
             },
             "status": {
-              "experimental": true,
+              "experimental": false,
```
Comment (caugner) [inline javascript/builtins/Promise.json:615]: @queengooborg I don't think https://github.com/mdn/browser-compat-data/blob/main/docs/data-guidelines/index.md#setting-experimental specifies what to do if one browser supports the feature, and another browser supports it only behind a flag. Should this remain experimental?

The first example may suggest that it should remain experimental for now:

> An API supported in Chrome and Firefox, without flags or prefixes.

```suggestion
              "experimental": true,
```

**Review by caugner — DISMISSED** [review]
I think it should remain experimental for now.

File: javascript/builtins/Promise.json (hunk 1 above)
Comment (chrisdavidmills) [reply javascript/builtins/Promise.json:615]: I think it should remain experimental. I'll commit this.

File: javascript/builtins/Promise.json [hunk 2]
```diff
@@ -584,7 +584,14 @@
...
               "firefox": {
-                "version_added": false
+                "version_added": "132",
+                "flags": [
+                  {
+                    "type": "preference",
+                    "name": "javascript.options.experimental.promise_try",
+                    "value_to_set": "true"
+                  }
+                ]
```
Comment (hamishwillee) [inline javascript/builtins/Promise.json:594]: @chrisdavidmills If this is in nightly, it gets done like this

```suggestion
                "version_added": "preview"
```

Comment (hamishwillee) [reply javascript/builtins/Promise.json:594]: Note, flag disappears.

Comment (chrisdavidmills) [reply javascript/builtins/Promise.json:594]: Ah, so "preview" is the word I was looking for. Thanks @hamishwillee!

Q: Why get rid of the flag? Surely that's important information that folks wanting to test the Fx implementation will want to know?

Comment (hamishwillee) [reply javascript/builtins/Promise.json:594]: Well you can just test on the release to find out the "non implemented version" and nightly to test the change.
Personally I am fine with the flag being there, and even see it as beneficial, but it isn't "needed".
Either way, "that's the BCD way"

**Review by hamishwillee — APPROVED** [review]

**Comment by hamishwillee** [comment]
Thanks!

**Pull request merged**